server/src/main/java/org/candlepin/controller/ModeManagerImpl.java

```diff
@@ -1,122 +0,0 @@
-/**
- * Copyright (c) 2009 - 2012 Red Hat, Inc.
- *
- * This software is licensed to you under the GNU General Public License,
- * version 2 (GPLv2). There is NO WARRANTY for this software, express or
- * implied, including the implied warranties of MERCHANTABILITY or FITNESS
- * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
- * along with this software; if not, see
- * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
- *
- * Red Hat trademarks are not licensed under GPLv2. No permission is
- * granted to use or replicate Red Hat trademarks that are incorporated
- * in this software or its documentation.
- */
-package org.candlepin.controller;
-
-import org.candlepin.common.config.Configuration;
-import org.candlepin.common.exceptions.SuspendedException;
-import org.candlepin.config.ConfigProperties;
-import org.candlepin.model.CandlepinModeChange;
-import org.candlepin.model.CandlepinModeChange.Mode;
-import org.candlepin.model.CandlepinModeChange.Reason;
-
-import com.google.inject.Inject;
-import com.google.inject.Singleton;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.xnap.commons.i18n.I18n;
-
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Date;
-import java.util.List;
-
-/**
- * This class drives Suspend Mode functionality. Suspend Mode is a
- * mode where Candlepin stops serving requests and only responds to
- * Status resource.
- *
- * @author fnguyen
- */
-@Singleton
-public class ModeManagerImpl implements ModeManager {
-    private static Logger log = LoggerFactory.getLogger(ModeManagerImpl.class);
-    private CandlepinModeChange cpMode = new CandlepinModeChange(
-        new Date(), Mode.NORMAL, Reason.STARTUP);
```

Comment:
I'm not really a fan of the idea of cycling the normal mode reasons, as it's not really useful for much beyond a "time since last incident" type of thing. As far as I can tell, the old mode reason thing was more an artifact of how the system worked rather than any by-design intent (especially considering it didn't work quite right when multiple suspend reasons got involved).

I could add the stack-based mode tracking, which would kind of address this, but the mode change reason would act more like a startup time than the old way of doing it. As such, I'm not sure if it's worth the effort to do unless someone actually checks those fields for normal mode state info and cared about the various reasons for restoring service.

Reply:
I'm fine with not adding this back, but I'd check if anyone (e.g. IT team) is using it for something.

Reply:
Preliminary check with IT and Sat suggests it is not needed today.